index.rst

```diff
@@ -1,29 +1,136 @@
+.. _sherlock: https://github.com/lsst-sqre/sherlock
+
+``exec:notebook``
+    Allows the user to spawn a lab in the Notebook Aspect.
+    This in turn allows arbitrary command execution within an unprivileged JupyterLab_ pod.
+
+.. _JupyterLab: https://jupyterlab.readthedocs.io/en/stable/
+
+``exec:portal``
+    Allows the user to perform operations in the Portal Aspect.
+    Due to the underlying queries the Portal performs, the user will also need ``read:image`` and ``read:tap`` scopes.
+
+``read:alertdb``
+    Grants access to receive alert packets and schemas from the alert archive database.
+
+``read:image``
+    Grants access to retrieve images accessible via the Science Platform.
+    Currently, this controls access to the HiPS (see DMTN-230_), SODA image cutout (see DMTN-208_), and DataLink (as implemented by datalinker_) services.
+
+    At present, there is a single scope for access to all images.
+    In the future, this may be broken into separate scopes for data releases or types of images.
+
+.. _DMTN-230: https://dmtn-230.lsst.io/
+.. _DMTN-208: https://dmtn-208.lsst.io/
+.. _datalinker: https://github.com/lsst-sqre/datalinker
+
+``read:tap``
+    Grants access to perform queries in the TAP service.
+
+``user:token``
+    Can create and modify tokens for the same user as the token that has this scope (as opposed to ``admin:token``, which allows any operation on tokens for any user).
+    This scope is automatically granted to users when they authenticate.
+    It exists as a separate scope primarily so that users can choose not to grant it to user tokens that they create, so that their programmatic tokens cannot themselves create new tokens.
+
+Creating new scopes
+===================
+
+Many authorization systems discover too late that they've allowed scopes to proliferate to the point where they become confusing and difficult to keep track of.
+To avoid this, the Rubin Science Platform only creates new scopes when there is a clear and compelling need.
+Specifically,
 
-Lists token scopes used by the authentication component of the Rubin Science Platform, defines them, and documents the services to which they grant access.
+#. there exist two users who should receive different levels of access in a way that cannot be represented by the existing scopes, and
+#. this access control difference must be done with scopes and not groups.
 
-Add content here
-================
+When to use scopes and when to use groups is fuzzy, but the general rule of thumb (as mentioned in :ref:`Purpose <purpose>` above) is that scopes control access to a service in its entirety, or to the administrative API as opposed to the user API of the service.
+Groups should be used for all other access control.
+Groups must be interpreted by the service (or by another service to which it delegates access control decisions).
+Scopes are enforced by the authentication layer, before the service ever sees the request.
```

Comment:
```suggestion
Scopes are enforced by the authentication layer, before the service ever sees the request, since they determine access to the service in the first place.
```